### ISSUE TITLE
Flatpak version

### ISSUE THREAD

**Issue by sramkrishna** [tomboy-notes/tomboy-ng#191]
I didn't even know about tomboy-ng until some spam hit the tomboy mailing list (you should archive that honestly and use the gnome discourse instead) and found out.

It's great that you're offering rpms and debs - but it would be really great if you could make a flatpak of tomboy-ng and offer it on flathub. What do you think?

**Comment by davidbannon** [comment]
Hi sramkrishna, thanks for your input.
Sorry, don't know what "gnome discourse" is.  In fact, tomboy-ng is possibly not Gnome's favourite son, unlike Tomboy, it does not require an install of GTK on Windows and Mac for example.  And I take every opportunity to take a swipe at Gnome3 and its campaign against the System Tray Icon.

Flatpak ? Honestly, I know nothing other than the broad idea. I use Ubuntu and personally hate the snaps that they insist are the next big thing. And I understand Snaps are Canonical's answer to Redhat's flatpak (or visa versa).  Both seem like a solution in need of a problem to solve IMHO.

Just why would you prefer a Flatpak sramkrishna ? tomboy-ng on Fedora (is that your chosen disro ?) only a couple of small dependencies, yum will install them before you know it !  And it shares nicely all those existing libraries, why install fresh, single user copies of all of them ?

Davo

( Not saying 'No' just really want to understand why ? )

**Comment by aguador** [comment]
Davo, certainly sramkrishna can tell you more about discourse, a social network site that some distros started using for their user fora (e.g., Manjaro).

I would be curious to know more about Flatpack as well as I have not seen the need for it. It might work for ng, but I follow the Evolution users mailing list and see nothing but problems caused by disappointed expectations as Flatpacks run in sandboxes. On the other hand, an AppImage would be a good way for people who simply wanted to try ng to test it out.

**Comment by davidbannon** [comment]
Yeah, AppImage.  The problem I see with all of them is that the implied bundling of everything needed into the image.  As far as I can tell, right now, almost all current distros have the GTK2 libraries installed "out of the box". But if I was to make a Snap, Flatpack or Appimage, I would have to cover the risk that somewhere, out there, there is a distro that does not. So, along with my 5meg binary, I would have to put 200meg of GTK2. 

And, I still don't see why. That is, how is an install based on an image better that a native install ?  The libraries I use are very standard, very predictable, very much the same on every system. The exception might relate to System Tray Icon, Debian and RedHat's Gnome 3 will not display it (Ubuntu and SUSE apply a patch that soles this problem).  But the image approach will not fix that, thats deep down in the desktop, way out of reach of the libraries in an image. 

I am still open to convincing arguments.

Davo

**Comment by chesty** [comment]
I often look for flatpaks first, then snaps, sometimes ppa's, and finally a distro version depending on the type of app and how actively it's developed.

If it's being updated by the developers regularly, the distro version is going to be old. That's sometimes good for software you really need working all the time, but for other apps living on the bleeding edge is fine. It's also possible to roll back an upgrade if it breaks. 

After I install tomboy via your instructions for debian, does it stay updated?

**Comment by davidbannon** [comment]
> After I install tomboy via your instructions for debian, does it stay updated?

No, people who want the newest have to manually install each time. tomboy-ng is under pretty active development, for awhile there, they had to do so (about) every month !  I guess I considered that a small effort compared to what I put into each release ....

However, I do take your point, while things are pretty stable now and I expect updates to come out a lot slower and to contain a lot less new content, it is time we had a better process. I most certainly do not update, manually, anything else on my system that frequently and its madness to expect my users to have to do so. 

1. There is another ticket open about getting tomboy-ng into the debian repos, I'll follow that up.  I think its mature and stable enough to make it into 'unstable'.
2. I'll look into a PPA, like the above, it will only help a subset of users.
3. I will cooperate in any way possible with anyone who wants to build snaps or flatpaks. Personally, I don't use them, therefore don't understand them and therefore, IMHO would be a bad person to manage such a sub project.  

I am hoping the only structural change in v0.3 will be the changes to how I deliver non-english help notes.  There a few fine tuning things happening and, I guess some bug fixes (sigh...).  But once that is done, I expect release cycles to become a lot longer and make that more extensive packaging practicable. Perhaps ! 

WRT item 3 above, if anyone wants to look into Snap/Flatpak, very welcome. Any model is possible ranging from someone working closely here to just grabbing the necessary files and delivering, with my blessing, packages to an appropriate repository.

Davo

**Comment by netlore** [comment]
Off topic, I know... but it may be relevant to someone!  20.04 removed python2, which also removed "createrepo", so you cannot manage RPM-MD repositories from Ubuntu 20.04....  there is a new C version of createrepo, which I have compiled and packaged, but it depends on new libraries which were not added until after 20.04, so.... since I do manage RPM-MD repo's for one of my projects... 20.04 broke my workflow, but 20.10 allowed me to fix it.... so, since I won't be downgrading, 20.10 it is for me.  I'm always on the 6 month cycle usually on the new releases ~6 weeks before release, and I have to say, 20.10 is looking excellent... and the 5.8 kernel fixed some outstanding issues for me, and at least 1 other person I know.

**Comment by davidbannon** [comment]
Makes sense !  When we get to 20.10 beta, I'll see if I can make an 'official' 20.10 -ng, should be easy given you have done my testing for me !

Davo

**Comment by davidbannon** [comment]
I don't think I want to add a FlatPack to my packaging list, its just a bit too long already !
tomboy-ng is now available in the Debian Repos, Bullseye, that will be the next Debian Stable soon. I use the word 'soon' carefully, Debian does not rush things.  As a flow on, it will also be in Ubuntu 21.04 and a number of other distros that base themselves on Debian.  
And I will continue to maintain the PPA for benefit of users who prefer an slower upgrade cycle (like me!).
So, think this is just about covered !
